web/src/main/webapp/rpc.ts

```diff
@@ -333,3 +333,32 @@ export class ZipReceiver extends Renderer<string> {
         rr.invoke(rec);
     }
 }
+
+// This collector waits for an operation to finish and calls a callback when
+// the operation has completed.
+export class CallbackCollector<T> extends Renderer<T> {
```

Comment:
I like this.
Maybe it's better to make this a base class which is extended by various renderers.
Let's call it something like OnCompleteRenderer: a renderer which only executes when everything is completed.